src/lib.rs

```diff
@@ -508,9 +510,12 @@ fn readline_edit<H: Helper>(
             }
             Cmd::EndOfFile => {
                 if !input_state.is_emacs_mode() && !s.line.is_empty() {
-                    s.edit_move_end()?;
                     break;
                 } else if s.line.is_empty() {
+                    // Move to end, in case cursor was in the middle of the
+                    // line, so that next thing application prints goes after
+                    // the input
+                    s.edit_move_buffer_end()?;
```

Comment:
If `line` is empty, there is no need to move the cursor, no?

Reply:
Yes. But we need to clear the hint. Fixed.